### DISCUSSION TITLE
[Deploy] Get Mia-Platform project id from deploy pipeline as default env

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#279] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v11

### Description

Hi guys,
a client needs to get the Mia-Platform `project id` (not the Gitlab Configurations id) from the deploy pipeline.
Is there a way to do that without manually creating an environment variable every time?

The client is developing, for a PoC, a custom deploy tool based on `mlp` and he needs to run, for example:
```bash
mlpPoC deploy --projectId ${MIA_PLATFORM_PROJECT_ID} ...
```

Is there a way to achieve this?

Thanks.

**Comment by davidebianchi** [comment, marked as answer]
Hi @silversoul93! At the moment, you could use the [webhook](https://docs.mia-platform.eu/docs/development_suite/webhooks-and-events/webhooks) to create it automatically.
At the moment, it's not possible to add other env to the trigger of the deploy.
Does this help you?

**Reply by frengooo** [reply]
Hi @davidebianchi, thanks for your reply.
How can I achieve this with webhooks? Should I create a custom microservice? Or just call the GitLab APIs to create the ENV?

**Reply by davidebianchi** [reply]
You should create a custom microservice, which should be called by the webhook at the project creation and create the environment variable on gitlab (if you use GitLab) for your GitLab project.

**Reply by frengooo** [reply]
Ok, thanks.
